internal/resources/managementclusterregistration/resource_management_cluster_registration.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package managementclusterregistration
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	clustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	managementclusterregistrationmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/managementclusterregistration"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func ResourceManagementClusterRegistration() *schema.Resource {
+	return &schema.Resource{
+		Schema: managementClusterRegistrationSchema,
+		//CreateContext: , TODO
+		//ReadContext: , TODO
+		//UpdateContext: , TODO
+		//DeleteContext: , TODO
+		Description: "Tanzu Mission Control Management Cluster Registration Resource",
+	}
+}
+
+var managementClusterRegistrationSchema = map[string]*schema.Schema{
+	NameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of this management cluster",
+		Required:    true,
+		ForceNew:    true,
+	},
+	common.MetaKey:     common.Meta,
+	specKey:            managementClusterRegistrationSpecSchema,
+	registerClusterKey: registerCluster,
```

Comment:
Will it be more readable to name it `registerTKGsvSphere` so that the user understands this schema only support TKGs and not TKGm?
cc: @vmw-vjn

Reply:
I see that the schema within `registerClusterKey` has fields with the name TKGm within. I think that should be good enough